crates/edr_provider/src/config.rs

```diff
@@ -28,3 +47,21 @@ pub struct AccountConfig {
     /// the balance of the account
     pub balance: U256,
 }
+
+impl Default for MemPoolConfig {
+    fn default() -> Self {
+        Self {
+            order: MineOrdering::Priority,
+        }
+    }
+}
+
+impl Default for MiningConfig {
+    fn default() -> Self {
+        Self {
+            auto_mine: true,
+            interval: IntervalConfig::Fixed(0),
```

Comment:
I regret modelling a disabled interval mining as an interval of 0. If it isn't a lot of work, what about adding a `Disabled` variant to the enum?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4619